mrp_bom_location/models/mrp_bom.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2017-20 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class MrpBom(models.Model):
+    _inherit = "mrp.bom"
+
+    location_id = fields.Many2one(
+        string="Location",
+        comodel_name="stock.location",
+        compute="_compute_location_id",
+        store=True,
+    )
+
+    @api.depends("picking_type_id")
+    def _compute_location_id(self):
+        for record in self:
+            if (
+                record.picking_type_id
+                and record.picking_type_id.default_location_src_id
+            ):
```

Comment:
```suggestion
            if record.picking_type_id.default_location_src_id:
```

it's the same, but shorter